x-pack/plugins/streams/server/routes/streams/list.ts

```diff
@@ -26,45 +25,59 @@ export const listStreamsRoute = createServerRoute({
     },
   },
   params: z.object({}),
-  handler: async ({ response, request, getScopedClients }) => {
+  handler: async ({
+    response,
+    request,
+    getScopedClients,
+  }): Promise<{ definitions: StreamDefinition[]; trees: StreamTree[] }> => {
     try {
       const { scopedClusterClient } = await getScopedClients({ request });
-      const definitions = await listStreams({ scopedClusterClient });
+      const { definitions } = await listStreams({ scopedClusterClient });
 
       const trees = asTrees(definitions);
 
-      return response.ok({ body: { streams: trees } });
+      return { definitions, trees };
```

Comment:
Why do you need the whole definition here? This will become a lot of stuff at some point.

Reply:
agree, but fine for now presumably?

Reply:
Sure not a blocker, I'm just wondering why you added it - the UI isn't rendering everything besides the id anyway, right? https://github.com/elastic/kibana/blob/b556194d229e990700699252011dcdc4fb64eed6/x-pack/plugins/streams_app/public/components/streams_table/index.tsx#L45

Is it because you didn't add the tree view yet and you wanted to avoid turning the tree back into a list in the UI again?

Reply:
Yeah, I'm just using the list now, and instinctively, I'd say the main listing API should return a flat list, and the tree is basically a NTH.

Reply:
Fine with that, I can fix this on my PR or we merge that one first and you resolve here, your call.